maintenance/troubleshoot/commands.md

```diff
@@ -0,0 +1,499 @@
+      hostPorts: Enabled
+      ipPools:
+      - blockSize: 26
+        cidr: 10.48.0.0/16
+        encapsulation: VXLANCrossSubnet
+        natOutgoing: Enabled
+        nodeSelector: all()
+      multiInterfaceMode: None
+      nodeAddressAutodetectionV4:
+        firstFound: true
+    cni:
+      ipam:
+        type: Calico
+      type: Calico
+```
+
+#### Run commands across multiple nodes
+
+Export THE_COMMAND_TO_RUN=date && for calinode in:
+
+```bash
+kubectl get pod -o wide -n calico-system | grep calico-node | awk '{print $1}'`; do echo $calinode; echo "-----"; 
+```
+
+```bash
+kubectl exec -n calico-system $calinode -- $THE_COMMAND_TO_RUN; printf "\n"; done calico-node-8xfmx
+```
+
+```
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+calico-node-9t8s7
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+calico-node-9cjhw
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+calico-node-cb7ff
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+calico-node-qoxvw
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+calico-node-wm5m2
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+```
+
+#### View pod info of pod that is not running
+
+```bash
+kubectl describe pods `<pod_name>`  -n `<namespace> `
+```
+
+```bash
+kubectl describe pods busybox -n default
+```
+
+```
+Events:
+  Type    Reason     Age   From                   Message
+  ----    ------     ----  ----                   -------
+  Normal  Scheduled  21s   default-scheduler      Successfully assigned default/busybox to ip-10-0-0-11
+  Normal  Pulling    20s   kubelet, ip-10-0-0-11  Pulling image "busybox"
+  Normal  Pulled     19s   kubelet, ip-10-0-0-11  Successfully pulled image "busybox"
+  Normal  Created    19s   kubelet, ip-10-0-0-11  Created container busybox
+  Normal  Started    18s   kubelet, ip-10-0-0-11  Started container busybox
+```
+
+#### View logs of a pod that is not running
+
+```bash
+kubectl logs `<pod_name>`  -n `<namespace>`
+```
+
+```bash
+kubectl logs busybox -n default
+```
+
+#### View kernel logs
+
+```bash
+journalctl -u kubelet
+```
+
+### Routing
+
+#### Verify routing table on the node
+
+Look for common issues like TCP port XXX is not open between instances.
+
+```bash
+ip route
+```
+
+```
+default via 10.0.0.1 dev eth0 proto dhcp src 10.0.0.10 metric 100 
+10.0.0.0/24 dev eth0 proto kernel scope link src 10.0.0.10 
+10.0.0.1 dev eth0 proto dhcp scope link src 10.0.0.10 metric 100 
+10.48.66.128/26 via 10.0.0.12 dev eth0 proto 80 onlink 
+10.48.231.0/26 via 10.0.0.11 dev eth0 proto 80 onlink 
+172.17.0.0/16 dev docker0 proto kernel scope link src 172.17.0.1 linkdown
+```
+
+#### Verify BGP peer status
```

Comment:
We have a new recommended way to do this: https://docs.projectcalico.org/reference/resources/caliconodestatus

We can leave this as-is for now to not add extra work, but we should follow up with an update for it.

Reply:
I created a backlog ticket to address this. Thanks.